dynamiqs/time_array.py

```diff
@@ -73,7 +79,7 @@ def _factory_callable(
             ' conversion at each solver time step.'
         )
 
-    return CallableTimeArray(x, f0)
+    return CallableTimeArray(Partial(f, args=args), f0)
```

Comment:
Maybe a line comment about this magic would be great (in the line on "Partial = jax way to do callable pytrees").